### ISSUE TITLE
SVM Plotting Decision Boundary Function

### ISSUE THREAD

**Issue by Praful932** [ageron/handson-ml2#252]
In the Chapter 5 Notebook, The plot decision boundary function, It is calculating the margin as `1/w[1]` shouldn't it be
`1/L2norm(w)` since the total margin width is `2/L2norm(w)`.

![image](https://user-images.githubusercontent.com/45713796/92909492-d06d5100-f444-11ea-97a1-25bef460bad6.png)

 I have tried with `LinearSVC` using l2 norm instead of `SVC(kernel="linear")`   and it gave a reasonable plot, is it that the margin calculation differs when using kernels

![image](https://user-images.githubusercontent.com/45713796/92910009-4a9dd580-f445-11ea-90aa-bbe5aa191dae.png)

**Comment by ibeauregard** [comment]
Let `decision_boundary = -w[0]/w1*x0 - b/w[1]`, as in the code above. Variable `decision_boundary` represents the x1 values for x0 at the decision boundary.

At the upper edge of the margin, we have `w[0]*x0 + w[1]*x1 + b = 1`, so `x1 = 1/w[1] - w[0]/w[1]*x1 - b/w[1]`.

Therefore, if we let `gutter_up` represent the x1 values for x0 at the upper edge of the margin, we can write `gutter_up = 1/w[1] - w[0]/w[1]*x1 - b/w[1]`.

Refering to the code in your screenshot, `gutter_up = decision_boundary + margin`, so we have to define margin as `margin = gutter_up - decision_boundary = 1/w[1] - w[0]/w[1]*x1 - b/w[1] - (-w[0]/w1*x0 - b/w[1]) = 1/w[1]`.

In other words, here `margin` is the **vertical** distance between the decision boundary and the edge of the margin, not the normal distance (which is 1).

**Comment by Praful932** [comment]
Ohh Thanks @ibeauregard , This made sense.
